src/validators/__tests__/is-nested.test.ts

```diff
@@ -73,4 +73,38 @@ describe('IsNested', () => {
         const errors = validateSync(obj);
         expect(errors).toHaveLength(0);
     });
+    describe('IsOneOf', () => {
+        it('does not allow more parameters than expected', () => {
+            class Fixture {
+                @IsNested({
+                    type: Child,
+                    hasXPropertiesDefined: 1,
+                })
```

Comment:
It's possible to [apply custom validators](https://github.com/Hippo-Analytics-Inc/policy-types/blob/master/src/dtos/insured/insured.dto.ts#L59) to specific fields without extending the decorator syntax. 

It's also possible to use [ValidateIf](https://github.com/typestack/class-validator#conditional-validation) to impose conditional validation. 

There's a good chance that your underlying goal can be achieved using `class-validator` directly, without extending this layer generically.